### PR TITLE
Unit tests

### LINKED ISSUE #72: add unit tests
#### Description

we are lacking in the department of unit tests and we need them

#### DoD (Definition of Done)

unit test coverage > 50%

#### Testing

Tests implemented

### PR DESCRIPTION
**Pull request by ViMuilu**
### Description

Unit tests. 
~~Vscode eslint plugin broke for some reason so some of the files might not be linted correctly.~~ Fixed

### Related issues
Fixes #72 

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Changes Made

Added unit tests

### Testing

- [x] Unit Tests

### CODE REVIEW

**Comment by blankdots** [comment]
can we run these files in github actions  and show coverage on coveralls ?
example: https://github.com/CSCfi/metadata-submitter-frontend/blob/master/.github/workflows/unittests.yml

**Comment by blankdots** [comment]
I added the `COVERALLS_REPO_TOKEN` so that example workflow should work here as well

File: src/components/AdvancedSearch.vue [hunk 1]
```diff
@@ -40,13 +42,15 @@
...
         <b-button
+          id="exampleButton"
           @click="exampleSearch"
           title="Insert example search parameters to the form"
           >Example range query</b-button
         ></span
       >
       <span id="basicSearch"
         ><b-button
+          id="basic"
```
Comment (blankdots) [inline src/components/AdvancedSearch.vue:53]: if you added these just for test you could have used: `data-testid` attribute to assign a value just for test

File: src/views/__tests__/Home.spec.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    const htmlAfterClick = wrapper.html();
+    expect(htmlAfterClick).not.toEqual(initialHtml);
+  });
+  it("Basic search button in advanced search changes form", async () => {
+    const basicSearchButton = wrapper.find("#basic");
+    const initialHtml = wrapper.html();
+    basicSearchButton.trigger("click");
+    await wrapper.vm.$nextTick();
+    const htmlAfterClick = wrapper.html();
+    //console.log(initialHtml)
```
Comment (blankdots) [inline src/views/__tests__/Home.spec.ts:32]: forgotten `console.log` ?

**Review by blankdots — APPROVED** [review]
maybe remove the `console.log` before merging, and think if you can use `data-testid` attribute

File: src/components/AdvancedSearch.vue (hunk 1 above)
Comment (ViMuilu) [reply src/components/AdvancedSearch.vue:53]: I'll change those

Comment (ViMuilu) [reply src/components/AdvancedSearch.vue:53]: The tests break if I change the id

Comment (blankdots) [reply src/components/AdvancedSearch.vue:53]: I think you need to do `const button = document.querySelector("[data-testid=id]");`

**Review by blankdots — APPROVED** [review]

**Pull request merged**